### PR TITLE
[Refactor] ForumThreadView - CreateThread and SearchResult refactor to use Twig

### LINKED ISSUE #3742: [Refactor] Twig engine for discussion forum
**What problem are you trying to solve with Submitty**
Currently, the Discussion Forum renders HTML directly from the View file. This has to be shifted to be using the Twig template engine for consistency with the rest of the Submitty project. 

**Describe the way you'd like to solve this problem**
Refactor the `app/views/forum/ForumThreadView.php` to use the Twig template engine and put the respective templates in the `app/templates/forum` directory.

### PR DESCRIPTION
**Pull request by xprilion**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Closes #3742 

### What is the new behavior?
The following use Twig templates - 

1. [x] CreateThread
2. [x] SearchResult
5. [x] DisplayThreadList - #3771 
6. [x] CreatePost - #3774 
3. [x] ShowForumThreads - #3780 
4. [x] GeneratePostList - #3774 
7. [x] StatPage - #3807 

### Other information?
Is this a breaking change? - None discovered yet.
How did you test - Not yet tested, only proper working checked manually. 

(This is a continuation to #3763 which was closed for shifting branch to Submitty repo, some previous conversation there.)

### CODE REVIEW

**Comment by andrewaikens87** [comment]
As @MasterOdin mentioned, I would agree that we should break this up into multiple PRs. For now, we can keep what you have done and I will review it. But for future work, we should make a separate PR.

**Comment by xprilion** [comment]
@andrewaikens87 Okay! I shall rename this PR to be more specific then! :)

File: site/app/views/forum/ForumThreadView.php [hunk 1]
```diff
@@ -997,97 +947,56 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
...
-								"forum_bar_buttons_left" => [],
-								"show_threads" => false,
-								"thread_exists" => $thread_exists,
-								"show_more" => false
-	]);
+        $this->core->getOutput()->addInternalJs('drag-and-drop.js');
+        $this->core->getOutput()->addVendorJs('flatpickr/flatpickr.js');
+        $this->core->getOutput()->addVendorJs('jquery.are-you-sure/jquery.are-you-sure.js');
 
+        $this->core->getOutput()->addInternalCss('flatpickr.min.css');
```
Comment (MasterOdin) [inline site/app/views/forum/ForumThreadView.php]: this should be `$this->core->getOutput()->addVendorCss('flatpickr/flatpickr.min.css');`

File: site/app/views/forum/ForumThreadView.php [hunk 2]
```diff
@@ -91,13 +48,16 @@ public function searchResult($threads){
 			$threadArray[$thread["thread_id"]][] = $thread;
 		}
 		$count = 1;
+
+		$thread_list = [];
+
 		foreach($threadArray as $thread_id => $data){
 			$thread_title = htmlentities($fromIdtoTitle[$thread_id], ENT_QUOTES | ENT_HTML5, 'UTF-8');
```
Comment (MasterOdin) [inline site/app/views/forum/ForumThreadView.php]: You don't need to do `htmlentities` here as Twig will take care of that for us.

File: site/app/views/forum/ForumThreadView.php [hunk 3]
```diff
@@ -997,97 +947,56 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
...
-					"show_thread_status" => true,
-					"show_announcement" => true,
-					"show_editcat" => true,
-					"submit_label" => "Submit Post",
-				]);
-			$return .= <<<HTML
-			</form>
-		</div>
-HTML;
+        if($this->core->getUser()->getGroup() <= 3){
```
Comment (andrewaikens87) [inline site/app/views/forum/ForumThreadView.php]: Please use `$this->core->getUser()->accessGrading()`

Comment (xprilion) [reply site/app/views/forum/ForumThreadView.php]: Done!

**Comment by xprilion** [comment]
Addressed all changes requested

**Pull request merged**